main.js

```diff
@@ -0,0 +1,286 @@
+            startGame.style.display = 'none';
+            startBtn.style.display = 'none';
+            playerTurn.innerText =`${playerOneNameInputVal}'s turn`;
+            playerTurn.style.display = 'inline';
+        };
+
+
+        return {
+            restartBtn,
+            startFunction,
+            resetDisplay,
+        }
+    })();
+
+    // create array from cells in order to grab index # of cell
+    function activateCells () {
+        cellElements.forEach(cell => {
+            cell.addEventListener('click', addToBoard, { once: true });
+        });
+    };
+
+    // create function to update board with player moves (x's and o's) upon click event
+    function addToBoard(e) {
+        let cell = e.target;
+        let selectedCell = cellArray.indexOf(cell);
+        cell.setAttribute('id', selectedCell);
+
+        if(x_turn) {
+            cell.innerText = player1.mark;
+            gameBoard.playerOneSelections.push(selectedCell);
+            x_turn = false;
+            playerTurn.innerText = `${player2name}'s turn`;
+            roundCount++
+            checkWin();
+            console.log(roundCount);
+
+        } else if (!x_turn) {
+            cell.innerText = player2.mark;
+            gameBoard.playerTwoSelections.push(selectedCell);
+            x_turn = true;
+            playerTurn.innerText = `${player1name}'s turn`;
+            roundCount++
+            checkWin();
+            console.log(roundCount);
+        };
+    };
+
+    function checkWin() {
+
+        let playerOneWin = false;
+        let playerTwoWin = false;
+        let winningLine;
+
+        function flatten(arr) {
+            let flatArray = [];
+
+            arr.forEach(element => {
+                if (Array.isArray(element)) {
+                    flatArray = flatArray.concat(flatten(element));
+                }
+                else {
+                    flatArray.push(element);
+                }
+            });
+            return flatArray;
+        };
+
+        const checkPlayerOneWin = (() => {
+            for (let i = 0; i < winningCombinations.length; i++) {
+                let newArray = flatten(winningCombinations[i]);
+                if(gameBoard.playerOneSelections.indexOf(newArray[0]) !== -1) {
+                    if(gameBoard.playerOneSelections.indexOf(newArray[1]) !== -1) {
+                        if(gameBoard.playerOneSelections.indexOf(newArray[2]) !== -1) {
+                            winningLine = winningCombinations[i];
+                            for (let i = 0; i < winningLine.length; i++) {
+                                document.getElementById(winningLine[i]).classList.toggle('winningLine');
+                            };
+                            playerOneWin = true;
+                            return;
+                        };
+                    };
+                } else {
+                    playerOneWin = false;
+                };
+            };
+            return;
+        })();
```

Comment:
Sections like this would be good to throw a comment or two into in order to explain what might be happening through these steps